core/src/Kokkos_View.hpp

```diff
@@ -264,14 +264,38 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
     // base class doesn't have constraint
     // FIXME: Eventually we need to deprecate this behavior and just use
     // BasicView implementation
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+    using l_t = typename mdspan_type::layout_type;
+    if (r >= rank()) {
```

Comment:
Warn?

Reply:
hmm since it is for deprecated code only I would not

Reply:
like runtime warning? I mean the deprecated behavior here is runtime value dependent.

Reply:
warn them that they are relying on something that is deprecated...